sessiontxn/isolation/base.go

```diff
@@ -68,12 +80,18 @@ func (p *baseTxnContextProvider) OnInitialize(ctx context.Context, tp sessiontxn
 	activeNow := true
 	switch tp {
 	case sessiontxn.EnterNewTxnDefault:
-		if err = p.sctx.NewTxn(ctx); err != nil {
+		if err := sessiontxn.CheckBeforeNewTxn(p.ctx, p.sctx); err != nil {
+			return err
+		}
+		if err := p.prepareTxn(false); err != nil {
 			return err
 		}
 	case sessiontxn.EnterNewTxnWithBeginStmt:
 		if !sessiontxn.CanReuseTxnWhenExplicitBegin(p.sctx) {
-			if err = p.sctx.NewTxn(ctx); err != nil {
+			if err := sessiontxn.CheckBeforeNewTxn(p.ctx, p.sctx); err != nil {
+				return err
+			}
+			if err := p.prepareTxnWhenSnapshotTSSet(); err != nil {
```

Comment:
If we do this here, the begin statement will use snapshotTS as startTS when `begin`?

Reply:
I guess the method name is misleading. It actually acquire a new ts when the snapshotTS is set. I will modify the name.